realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -486,27 +503,23 @@ private static void runInitializationCallback(long nativeSharedRealmPtr, OsRealm
     /**
      * Called from JNI when the partial sync callback is invoked from the ObjectStore.
      * @param error if the partial sync query failed to register.
-     * @param resultsNativePtr pointer to the {@code Results} of the partial sync query.
-     * @param tableName the table name for the {@code Results}.
-     * @param clazz the type used with the partial sync query to obtain the {@code Results}.
+     * @param nativeResultsPtr pointer to the {@code Results} of the partial sync query.
      * @param callback the callback registered from the user to notify the success/error of the partial sync query.
-     * @param realm the Realm instance to be used to create the {@link RealmResults} (in case of success).
      */
     @SuppressWarnings("unused")
-    private static void runPartialSyncRegistrationCallback(@Nullable String error, long resultsNativePtr, String tableName, Class clazz, Realm.PartialSyncCallback callback, Realm realm) {
-        // don't notify success or error if the callback/realm has been GC'ed
-        if (callback != null && realm != null) {
-            if (error != null) {
-                callback.onError(new RealmException(error));
-            } else {
-                Table table = realm.sharedRealm.getTable(tableName);
-                Collection collection = new Collection(realm.sharedRealm, table, resultsNativePtr, true);
-                RealmResults results = new RealmResults(realm, collection, clazz);
-                callback.onSuccess(results);
-            }
+    private static void runPartialSyncRegistrationCallback(@Nullable String error, long nativeResultsPtr,
+                                                           PartialSyncCallback callback) {
+        if (error != null) {
+            callback.onError(new RealmException(error));
+        } else {
+            @SuppressWarnings("ConstantConditions")
+            Table table = callback.sharedRealm.getTable(Table.getTableNameForClass(callback.className));
```

Comment:
what about making `runPartialSyncRegistrationCallback` non static, we will no longer need to keep `sharedRealm` in the callback

Reply:
I will change `PartialSyncCallback.sharedRealm` to be a type of `WeakReference<SharedRealm>` instead. That is easier to check.